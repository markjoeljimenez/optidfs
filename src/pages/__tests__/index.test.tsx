import '@testing-library/jest-dom';
import { rest } from 'msw';
import { setupServer } from 'msw/node';
import { mockNextUseRouter } from '../../scripts/testing/util/mockNextUseRouter';
import { render, screen, waitFor } from '../../scripts/testing/render';
import userEvent from '@testing-library/user-event';
import Index from '../index';

const server = setupServer(
	rest.get('http://127.0.0.1:5000', (req, res, ctx) => {
		return res(
			ctx.json([
				{
					fullName: 'Football',
					hasPublicContests: true,
					isEnabled: true,
					positions: [
						'QB',
						'RB',
						'RB',
						'WR',
						'WR',
						'WR',
						'TE',
						'FLEX',
						'DST',
					],
					regionAbbreviatedSportName: 'NFL',
					regionFullSportName: 'Football',
					sortOrder: 1,
					sportId: 1,
					supported: true,
				},
				{
					fullName: 'Basketball',
					hasPublicContests: true,
					isEnabled: true,
					positions: ['PG', 'SG', 'SF', 'PF', 'C', 'G', 'F', 'UTIL'],
					regionAbbreviatedSportName: 'NBA',
					regionFullSportName: 'Basketball',
					sortOrder: 2,
					sportId: 4,
					supported: true,
				},
			])
		);
	})
);

beforeAll(() => server.listen({ onUnhandledRequest: 'error' }));
afterEach(() => server.resetHandlers());
afterAll(() => server.close());

describe('Index', () => {
	mockNextUseRouter({
		route: '/',
		pathname: '/',
		query: '',
		asPath: ``,
	});

	it('renders index page successfully', () => {
		const dashboard = render(<Index />);

		expect(dashboard).toMatchSnapshot();
	});

	it('should proceed to step 2 if provider and sport is selected', async () => {
		render(<Index />);

		const providerSelect = screen.getByTestId(
			'provider-select'
		) as HTMLSelectElement;
		const sportSelect = screen.getByTestId(
			'sports-select'
		) as HTMLSelectElement;
		const nextButton = screen.getByTestId(
			'to-next-step'
		) as HTMLButtonElement;

		expect(sportSelect.hasAttribute('disabled')).toBe(true);
		expect(nextButton.hasAttribute('disabled')).toBe(true);

		// Select provider
		userEvent.selectOptions(providerSelect, ['yahoo']);

		await waitFor(() =>
			expect(sportSelect.hasAttribute('disabled')).toBe(false)
		);

		expect(nextButton.hasAttribute('disabled')).toBe(true);

		// Select sport
		userEvent.selectOptions(sportSelect, ['4']);

		await waitFor(() =>
			expect(nextButton.hasAttribute('disabled')).toBe(false)
		);

		// Click button to proceed to step 2
		userEvent.click(nextButton);

		expect(providerSelect).not.toBeInTheDocument();
		expect(sportSelect).not.toBeInTheDocument();
		expect(nextButton).not.toBeInTheDocument();
	});
});
