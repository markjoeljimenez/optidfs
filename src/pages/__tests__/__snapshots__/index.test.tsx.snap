// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index renders index page successfully 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex flex-col items-center justify-center"
        data-step="1"
      >
        <div
          class="flex-1"
        >
          <div
            class="space-y-3"
          >
            <div
              class="text-center"
            >
              <h1
                class="text-3xl"
              >
                Welcome!
              </h1>
              <p>
                Start by selecting a DFS provider and sport
              </p>
            </div>
            <div
              class="flex space-x-3 items-center"
            >
              <div>
                <div
                  class=""
                >
                  <label
                    class="block text-sm font-medium text-gray-700 sr-only"
                    for="selectProvider"
                  >
                    Select provider
                  </label>
                </div>
                <div
                  class="flex"
                >
                  <select
                    class="focus:ring-indigo-500 focus:border-indigo-500 block w-full sm:text-sm border-gray-300 rounded cursor-pointer"
                    data-testid="provider-select"
                    id="selectProvider"
                  >
                    <option
                      disabled=""
                      value=""
                    >
                      Select provider
                    </option>
                    <option
                      data-testid="draftkings"
                      value="draftkings"
                    >
                      DraftKings
                    </option>
                    <option
                      data-testid="yahoo"
                      value="yahoo"
                    >
                      Yahoo
                    </option>
                  </select>
                </div>
              </div>
              <div
                class="relative"
              >
                <div>
                  <div
                    class=""
                  >
                    <label
                      class="block text-sm font-medium text-gray-700 sr-only"
                      for="selectSport"
                    >
                      Select sport
                    </label>
                  </div>
                  <div
                    class="flex"
                  >
                    <select
                      class="focus:ring-indigo-500 focus:border-indigo-500 block w-full sm:text-sm border-gray-300 rounded cursor-pointer"
                      data-testid="sports-select"
                      disabled=""
                      id="selectSport"
                    >
                      <option
                        disabled=""
                        value=""
                      >
                        Select sport
                      </option>
                    </select>
                  </div>
                </div>
              </div>
              <button
                class="text-white rounded-full  bg-gray-300 cursor-default"
                data-testid="to-next-step"
                disabled=""
              >
                <div
                  class=""
                  style="width: 30px; height: 30px;"
                >
                  <svg
                    class="fill-current"
                    viewBox="0 0 24 24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <g>
                      <path
                        d="M10.5 17a1 1 0 0 1-.71-.29 1 1 0 0 1 0-1.42L13.1 12 9.92 8.69a1 1 0 0 1 0-1.41 1 1 0 0 1 1.42 0l3.86 4a1 1 0 0 1 0 1.4l-4 4a1 1 0 0 1-.7.32z"
                      />
                    </g>
                  </svg>
                </div>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex flex-col items-center justify-center"
      data-step="1"
    >
      <div
        class="flex-1"
      >
        <div
          class="space-y-3"
        >
          <div
            class="text-center"
          >
            <h1
              class="text-3xl"
            >
              Welcome!
            </h1>
            <p>
              Start by selecting a DFS provider and sport
            </p>
          </div>
          <div
            class="flex space-x-3 items-center"
          >
            <div>
              <div
                class=""
              >
                <label
                  class="block text-sm font-medium text-gray-700 sr-only"
                  for="selectProvider"
                >
                  Select provider
                </label>
              </div>
              <div
                class="flex"
              >
                <select
                  class="focus:ring-indigo-500 focus:border-indigo-500 block w-full sm:text-sm border-gray-300 rounded cursor-pointer"
                  data-testid="provider-select"
                  id="selectProvider"
                >
                  <option
                    disabled=""
                    value=""
                  >
                    Select provider
                  </option>
                  <option
                    data-testid="draftkings"
                    value="draftkings"
                  >
                    DraftKings
                  </option>
                  <option
                    data-testid="yahoo"
                    value="yahoo"
                  >
                    Yahoo
                  </option>
                </select>
              </div>
            </div>
            <div
              class="relative"
            >
              <div>
                <div
                  class=""
                >
                  <label
                    class="block text-sm font-medium text-gray-700 sr-only"
                    for="selectSport"
                  >
                    Select sport
                  </label>
                </div>
                <div
                  class="flex"
                >
                  <select
                    class="focus:ring-indigo-500 focus:border-indigo-500 block w-full sm:text-sm border-gray-300 rounded cursor-pointer"
                    data-testid="sports-select"
                    disabled=""
                    id="selectSport"
                  >
                    <option
                      disabled=""
                      value=""
                    >
                      Select sport
                    </option>
                  </select>
                </div>
              </div>
            </div>
            <button
              class="text-white rounded-full  bg-gray-300 cursor-default"
              data-testid="to-next-step"
              disabled=""
            >
              <div
                class=""
                style="width: 30px; height: 30px;"
              >
                <svg
                  class="fill-current"
                  viewBox="0 0 24 24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <g>
                    <path
                      d="M10.5 17a1 1 0 0 1-.71-.29 1 1 0 0 1 0-1.42L13.1 12 9.92 8.69a1 1 0 0 1 0-1.41 1 1 0 0 1 1.42 0l3.86 4a1 1 0 0 1 0 1.4l-4 4a1 1 0 0 1-.7.32z"
                    />
                  </g>
                </svg>
              </div>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "store": Object {
    "@@observable": [Function],
    "dispatch": [Function],
    "getState": [Function],
    "replaceReducer": [Function],
    "subscribe": [Function],
  },
  "unmount": [Function],
}
`;
